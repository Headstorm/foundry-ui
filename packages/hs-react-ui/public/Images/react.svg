<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 19.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 width="561.8px" height="499.9px" viewBox="-80.4 151.2 561.8 499.9" style="enable-background:new -80.4 151.2 561.8 499.9;"
	 xml:space="preserve">
<style type="text/css">
	.st0{fill:#939598;}
</style>
<circle class="st0" cx="200.5" cy="400.6" r="50.2"/>
<path class="st0" d="M200.5,515.6c-70.5,0-132.1-8.3-178.2-24.1c-29.9-10.2-55.3-23.8-73.4-39.3c-19.2-16.4-29.4-34.3-29.4-51.6
	c0-33.2,36.4-65.7,97.5-86.9c50-17.4,115.2-27.1,183.4-27.1c67,0,131.3,9.4,181,26.6c29.1,10,53.6,23,71,37.4
	c18.9,15.8,28.9,33.1,28.9,50c0,34.5-40.7,69.4-106.3,91.1C328.6,507.1,266.6,515.6,200.5,515.6z M200.5,310.6
	c-64.7,0-128.7,9.4-175.5,25.7c-40.5,14.1-81.4,39.5-81.4,64.3c0,10,7.4,21.8,21,33.4c15.8,13.5,38.4,25.6,65.5,34.8
	c43.6,14.9,102.6,22.8,170.4,22.8c63.6,0,122.9-8,167-22.7c61.7-20.5,89.9-49.8,89.9-68.3c0-9.5-7.2-20.7-20.3-31.6
	c-15.1-12.6-37.1-24.1-63.4-33.2C326.4,319.6,264.9,310.6,200.5,310.6z"/>
<path class="st0" d="M86.6,650.8c-10.2,0-19.2-2.2-26.8-6.6C31,627.6,21.1,579.8,33.1,516.3c9.9-52.1,34.1-113.3,68.2-172.4h0
	c33.5-58,73.7-109,113.4-143.5c23.2-20.2,46.7-35,67.9-42.8c23.1-8.5,43.1-8.5,57.7-0.1c29.9,17.2,39.8,70,25.8,137.6
	c-9.9,48-33.5,105.9-66.5,163.2c-35.2,61-73.2,110.2-109.9,142.3c-23.8,20.8-48.3,36-70.7,43.9C107.4,648.7,96.4,650.8,86.6,650.8z
	 M122.1,355.9c-32.3,56-56.2,116.1-65.4,164.9c-8,42.1-6.5,90.2,15,102.6c16.1,9.3,55,0.5,102.3-40.9c34.7-30.3,71-77.5,104.9-136.2
	c31.8-55.1,54.4-110.5,63.8-156c6-29.1,7.3-55.5,3.8-76.4c-3-17.9-9.4-30.6-18.2-35.6c-8.2-4.7-21.5-4.1-37.5,1.8
	c-18.5,6.8-39.4,20.1-60.4,38.4C192.8,251.3,154.3,300.1,122.1,355.9L122.1,355.9z"/>
<path class="st0" d="M314.4,651.1c-27.2,0-61.7-16.4-97.7-47.4c-40.2-34.6-81.1-86.1-115.3-145.2v0c-33.6-58-57.6-118.3-67.7-170
	c-5.9-30.2-7-57.9-3.2-80.2c4.2-24.3,14.1-41.6,28.8-50.1c29.8-17.3,80.5,0.5,132.1,46.4c36.6,32.5,75,81.9,108.1,139.1
	c35.3,61,59,118.5,68.4,166.3c6.1,31,7.1,59.8,2.8,83.2c-4.6,24.9-15,42.6-30,51.3C333.2,648.9,324.3,651.1,314.4,651.1z
	 M122.2,446.5c32.4,56,72.6,106.7,110.2,139c32.5,28,74.9,50.6,96.4,38.2c16.1-9.3,27.9-47.4,15.7-109
	c-9-45.2-31.7-100.2-65.6-158.9c-31.9-55.1-68.6-102.4-103.3-133.2c-22.2-19.7-44.4-34.1-64.2-41.5c-17-6.4-31.2-7.1-39.9-2.1
	c-8.2,4.7-14.3,16.6-17.2,33.4c-3.3,19.4-2.3,44.2,3.1,71.5C66.9,333,89.9,390.7,122.2,446.5L122.2,446.5z"/>
</svg>
