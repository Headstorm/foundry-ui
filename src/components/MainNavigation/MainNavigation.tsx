import React, { ReactNode } from 'react';
import styled from 'styled-components';
import { getFontColorFromVariant, disabledStyles } from '../../utils/color';
import { Div } from '../../htmlElements';
import { StyledSubcomponentType, SubcomponentPropsType } from '../commonTypes';
import { useTheme } from '../../context';
import Button from '../Button';
import Text from '../Text';
import variants from '../../enums/variants';

export type HideAnimationPropType = {
  length?: number;
  origin?: string;
  collapsed: boolean;
};

export const defaultHideAnimation = ({
  length = 0.1,
  origin = 'top',
  collapsed,
}: HideAnimationPropType) => `
    transform: ${collapsed ? 'scaleY(0)' : 'scaleY(1)'};
    transform-origin: ${origin};
    transition: transform ${length}s cubic-bezier(0, .7, .9, 1);;    
`;

export const Container = styled(Div)`
  ${({
    color,
    height,
    position,
    location,
    animation,
    disabled,
  }: {
    color: string;
    height: string;
    position: string;
    location: string;
    animation: () => void;
    disabled: boolean;
  }) => `
    position: ${position};
    ${location}
    background-color: ${color};
    color: ${getFontColorFromVariant(variants.fill, color)};
    height: ${height};
    width: 100%;
    ${animation}    
    ${disabled ? disabledStyles() : ''}
  `}
`;

const NavFlex = styled(Div)`
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  align-content: stretch;
  padding-left: 1rem;
`;

export const NavSection = styled(Div)`
  ${({ bodyBelow }: { bodyBelow: boolean }) => `
    flex-direction: ${bodyBelow ? 'column' : 'row'};
    align-items: ${bodyBelow ? 'flex-start' : 'center'};
  `}
  display: flex;
  height: 100%;
`;

export const Header = styled(NavSection)`
  ${({ bodyBelow }: { bodyBelow: boolean }) => `
    align-self: ${bodyBelow ? 'flex-start' : 'center'};
  `}
  margin-right: 1rem;
  width: fit-content;
`;

export const Body = styled(NavSection)`
  width: 100%;
`;

export const NavButtonContainer = styled(Button.Container)`
  width: fit-content;
  padding: 0.25rem 0rem 0.25rem 1rem;
  height: 100%;
  border-radius: 0 0 0 0;
  &:hover {
    text-decoration: underline;
  }
  &:focus {
    box-shadow: 0 0 0;
  }
`;

export const Footer = styled(NavSection)`
  margin-left: auto;
`;

export type NavButton = {
  label: string;
  onClick: () => void;
};

const getGetScrollSpeed = (scrollSpeed = 50) => {
  let lastPos: number | null;
  let newPos: number | null;
  let timer: NodeJS.Timeout;
  let delta: number;
  const delay = scrollSpeed; // in "ms" (higher means lower fidelity )

  function clear() {
    lastPos = null;
    delta = 0;
  }

  clear();

  return () => {
    newPos = window.scrollY;
    if (lastPos != null) {
      delta = newPos - lastPos;
    }
    lastPos = newPos;
    clearTimeout(timer);
    timer = setTimeout(clear, delay);
    return delta;
  };
};

export const getScrollPosition = (): number => {
  const doc = document.documentElement;
  return (window.pageYOffset || doc.scrollTop) - (doc.clientTop || 0);
};

export interface MainNavigationProps {
  StyledContainer?: StyledSubcomponentType;
  StyledHeader?: StyledSubcomponentType;
  StyledBody?: StyledSubcomponentType;
  StyledFooter?: StyledSubcomponentType;

  containerProps?: SubcomponentPropsType;
  headerProps?: SubcomponentPropsType;
  bodyProps?: SubcomponentPropsType;
  footerProps?: SubcomponentPropsType;
  navButtonProps?: SubcomponentPropsType[];

  containerRef?: React.RefObject<HTMLDivElement>;
  headerRef?: React.RefObject<HTMLDivElement>;
  bodyRef?: React.RefObject<HTMLDivElement>;
  footerRef?: React.RefObject<HTMLDivElement>;
  navButtonRefs?: React.RefObject<HTMLButtonElement>[];

  header?: ReactNode;
  body?: ReactNode;
  navButtons?: NavButton[];
  labelFontSize?: string;
  footer?: ReactNode;

  hidden?: boolean;
  disabled?: boolean;
  hideBody?: boolean;
  bodyBelow?: boolean;
  hiddenBelowY?: number;
  onScroll?: () => void;
  HideAnimationProps?: HideAnimationPropType;
  hideAnimation?: (value: HideAnimationPropType) => void;
  // set CSS position type ie `relative`, `absolute`, `static`, etc.
  position?: string;
  // set CSS location directly like `top: 10px; left: 10px;`
  location?: string;
  height?: string;
  color?: string;
}

const MainNavigation = ({
  StyledContainer = Container,
  StyledHeader = Header,
  StyledBody = Body,
  StyledFooter = Footer,

  containerProps = {},
  headerProps = {},
  bodyProps = {},
  footerProps = {},
  navButtonProps = [{}],

  containerRef,
  headerRef,
  bodyRef,
  footerRef,
  navButtonRefs = [],

  header,
  body,
  navButtons,
  labelFontSize = '1rem',
  footer,

  hidden = false,
  disabled = false,
  hideBody = false,
  bodyBelow = false,
  // auto hiding below Y uses window.onscroll by default, for server-side rendering use your own onscroll to set the hidden argument
  hiddenBelowY,
  onScroll,
  HideAnimationProps,
  hideAnimation = defaultHideAnimation,
  position = 'relative',
  location = '',
  height = 'fit-content',
  color,
}: MainNavigationProps): JSX.Element => {
  const { colors } = useTheme();
  const backgroundColor = color || colors.primaryDark;
  const [isHidden, setIsHidden] = React.useState(false);
  if (typeof hiddenBelowY !== 'undefined') {
    window.onscroll = () => {
      if (onScroll) {
        onScroll();
      }
      setIsHidden(getScrollPosition() > hiddenBelowY);
    };
  }
  const animationProps = HideAnimationProps || {
    collapsed: hidden || isHidden,
  };

  return (
    <StyledContainer
      ref={containerRef}
      height={height}
      color={backgroundColor}
      position={position}
      location={location}
      animation={hideAnimation(animationProps)}
      disabled={disabled}
      {...containerProps}
    >
      {bodyBelow ? (
        <>
          <NavFlex>
            {header && (
              <StyledHeader ref={headerRef} {...headerProps}>
                {header}
              </StyledHeader>
            )}
            {footer && (
              <StyledFooter ref={footerRef} {...footerProps}>
                {footer}
              </StyledFooter>
            )}
          </NavFlex>
          {!hideBody ? (
            <StyledBody ref={bodyRef} bodyBelow={bodyBelow} {...bodyProps}>
              {navButtons &&
                navButtons.map((navButton, index) => (
                  <Button
                    containerRef={navButtonRefs[index]}
                    key={navButton.label}
                    onClick={navButton.onClick}
                    color="#0000"
                    StyledContainer={NavButtonContainer}
                    containerProps={{ ...navButtonProps[index] }}
                  >
                    <Text
                      color={getFontColorFromVariant(variants.fill, backgroundColor)}
                      size={labelFontSize}
                    >
                      {navButton.label}
                    </Text>
                  </Button>
                ))}
              {body && body}
            </StyledBody>
          ) : (
            ''
          )}
        </>
      ) : (
        <NavFlex>
          {header && (
            <StyledHeader ref={headerRef} {...headerProps}>
              {header}
            </StyledHeader>
          )}
          {!hideBody ? (
            <StyledBody ref={bodyRef} bodyBelow={bodyBelow} {...bodyProps}>
              {navButtons &&
                navButtons.map((navButton, index) => (
                  <Button
                    containerRef={navButtonRefs[index]}
                    key={navButton.label}
                    onClick={navButton.onClick}
                    color="#0000"
                    StyledContainer={NavButtonContainer}
                    containerProps={{ ...navButtonProps[index] }}
                  >
                    <Text
                      color={getFontColorFromVariant(variants.fill, backgroundColor)}
                      size={labelFontSize}
                    >
                      {navButton.label}
                    </Text>
                  </Button>
                ))}
              {body && body}
            </StyledBody>
          ) : (
            ''
          )}
          {footer && (
            <StyledFooter ref={footerRef} {...footerProps}>
              {footer}
            </StyledFooter>
          )}
        </NavFlex>
      )}
    </StyledContainer>
  );
};

MainNavigation.Container = Container;
MainNavigation.NavButtonContainer = NavButtonContainer;
MainNavigation.Header = Header;
MainNavigation.Body = Body;
MainNavigation.Footer = Footer;
// if using server-side rendering these will not work, since they use window.location, and should be handled outside the component
MainNavigation.getScrollSpeed = getGetScrollSpeed;
MainNavigation.getScrollPosition = getScrollPosition;

export default MainNavigation;
