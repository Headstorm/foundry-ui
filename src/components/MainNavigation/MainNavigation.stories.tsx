import React from 'react';
import { Story, Meta } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import styled from 'styled-components';
import Icon from '@mdi/react';
import * as Icons from '@mdi/js';
import colors from '../../enums/colors';
// import { Main } from '../../htmlElements';
import MainNavigation, { MainNavigationProps } from './MainNavigation';
import Text from '../Text';
import TextInput from '../TextInput';
import Button from '../Button';

type DefaultProps = MainNavigationProps;

const App = styled.div`
  outline: 1px grey solid;
`;

const HeaderText = styled(Text.Container)`
  white-space: nowrap;
`;

const HamburgerButton = styled(Button.Container)`
  height: 100%;
  padding: 0.5rem;
  margin-right: 1rem;
  &:focus {
    box-shadow: 0 0 0;
  }
`;

const SearchBar = styled(TextInput.Container)`
  ${({ expanded }: { expanded: boolean }) => `
    width: ${expanded ? '99%' : '80%'};
    margin-left: auto;
    margin-right: ${expanded ? 'auto' : '1rem'};
    margin-top: 0.25rem;
    margin-bottom: 0.25rem;
    &:focus-within {
      outline: none;
      box-shadow: 0 0 5px 0.15rem ${colors.tertiary};
    }
  `}
`;

const SearchInput = styled(TextInput.Input)`
  &:focus {
    box-shadow: 0 0 0;
  }
`;

const lorem = `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
ut labore et dolore magna aliqua. Porttitor massa id neque aliquam vestibulum morbi
blandit cursus risus. Mi bibendum neque egestas congue quisque egestas diam. Pulvinar
sapien et ligula ullamcorper. Ac turpis egestas integer eget aliquet nibh. Bibendum at
varius vel pharetra vel turpis nunc eget. Interdum varius sit amet mattis vulputate enim
nulla aliquet porttitor. Morbi tempus iaculis urna id volutpat. Auctor neque vitae tempus
quam pellentesque. Tempor commodo ullamcorper a lacus vestibulum sed. Malesuada proin
libero nunc consequat interdum varius. At volutpat diam ut venenatis tellus in metus
vulputate eu. Quam adipiscing vitae proin sagittis nisl. Sagittis purus sit amet volutpat
consequat. Magna fringilla urna porttitor rhoncus dolor. Vitae aliquet nec ullamcorper sit
amet. Imperdiet dui accumsan sit amet nulla facilisi. Sapien faucibus et molestie ac
feugiat sed lectus vestibulum. Sed tempus urna et pharetra pharetra. Ut placerat orci
nulla pellentesque dignissim enim sit amet. Netus et malesuada fames ac turpis egestas
integer. Bibendum est ultricies integer quis auctor elit sed. Et netus et malesuada fames
ac turpis. Ultrices sagittis orci a scelerisque purus semper eget duis at. Non blandit
massa enim nec dui nunc. Felis bibendum ut tristique et egestas quis. Volutpat ac
tincidunt vitae semper. Adipiscing vitae proin sagittis nisl rhoncus mattis rhoncus urna.
Donec enim diam vulputate ut pharetra sit amet aliquam. Quis enim lobortis scelerisque
fermentum dui faucibus in ornare quam. Suspendisse sed nisi lacus sed viverra tellus in.
Proin nibh nisl condimentum id venenatis. Duis tristique sollicitudin nibh sit amet
commodo nulla. Sed cras ornare arcu dui. Consectetur adipiscing elit pellentesque habitant
morbi tristique senectus et netus. Ut etiam sit amet nisl purus in mollis nunc. Aliquam id
diam maecenas ultricies mi eget mauris pharetra et. Neque volutpat ac tincidunt vitae
semper quis lectus. Amet nisl suscipit adipiscing bibendum. Ultrices sagittis orci a
scelerisque purus. Et malesuada fames ac turpis egestas sed tempus. Euismod lacinia at
quis risus sed vulputate. Egestas fringilla phasellus faucibus scelerisque eleifend donec.
Nulla posuere sollicitudin aliquam ultrices sagittis orci a scelerisque purus. Nunc
lobortis mattis aliquam faucibus purus in massa tempor. Euismod nisi porta lorem mollis
aliquam. Varius vel pharetra vel turpis nunc eget lorem dolor sed. Tempus urna et pharetra
pharetra massa massa ultricies mi quis. Aliquam faucibus purus in massa. Pellentesque sit
amet porttitor eget dolor. Sit amet cursus sit amet dictum sit amet. Adipiscing elit ut
aliquam purus sit. Velit ut tortor pretium viverra suspendisse. Aliquet nibh praesent
tristique magna sit amet purus gravida. Risus pretium quam vulputate dignissim suspendisse
in est ante. Nulla pharetra diam sit amet nisl suscipit adipiscing bibendum est. Tristique
et egestas quis ipsum suspendisse ultrices gravida. Aliquam ut porttitor leo a diam. Quis
hendrerit dolor magna eget est lorem. Sed tempus urna et pharetra pharetra massa massa
ultricies mi. Venenatis urna cursus eget nunc scelerisque viverra mauris. Sed egestas
egestas fringilla phasellus faucibus scelerisque eleifend. Venenatis tellus in metus
vulputate eu. Parturient montes nascetur ridiculus mus. Id diam vel quam elementum
pulvinar etiam. Magna etiam tempor orci eu.`;

const getIconPath = (path: string, color: string) =>
  path ? <Icon color={color} size="1rem" path={path} /> : undefined;

export const Default: Story<DefaultProps> = ({ ...args }: DefaultProps) => {
  const [hidden, setHidden] = React.useState(args.hidden);
  React.useEffect(() => setHidden(args.hidden), [args.hidden]);
  const [expanded, setExpanded] = React.useState(args.bodyBelow);
  React.useEffect(() => setExpanded(args.bodyBelow), [args.bodyBelow]);
  const [active, setActive] = React.useState(0);

  const getScrollSpeed = MainNavigation.getScrollSpeed(50);

  return (
    <App>
      <MainNavigation
        hidden={hidden}
        hideBody={args.hideBody && !expanded}
        hiddenBelowY={args.hiddenBelowY}
        bodyBelow={expanded}
        onScroll={() => {
          action(`Scrolled with speed: ${Math.abs(getScrollSpeed())}`)();
        }}
        position={args.position}
        location={args.location}
        color={args.color}
        height={args.height}
        header={args.header}
        body={
          <TextInput
            placeholder="Search Bar"
            StyledContainer={SearchBar}
            containerProps={{ expanded }}
            StyledInput={SearchInput}
            iconPrefix={getIconPath(Icons.mdiMagnify, 'grey')}
          />
        }
        navButtons={[
          {
            label: 'Link 1',
            onClick: () => {
              setActive(0);
              action('click 1')();
            },
          },
          {
            label: 'Link 2',
            onClick: () => {
              setActive(1);
              action('click 2')();
            },
          },
          {
            label: 'Link 3',
            onClick: () => {
              setActive(2);
              action('click 3')();
            },
          },
        ]}
        activeButton={active}
        labelFontSize={args.labelFontSize}
        footer={
          <Button
            color="#0000"
            StyledContainer={HamburgerButton}
            onClick={() => setExpanded(!expanded)}
          >
            {getIconPath(Icons.mdiMenu, 'white')}
          </Button>
        }
      />
      <body>
        <h2>
          <Text>Lorem ipsum</Text>
        </h2>
        <Text>{lorem.repeat(10)}</Text>
      </body>
    </App>
  );
};
Default.args = {
  hidden: false,
  hiddenBelowY: 100,
  hideBody: false,
  bodyBelow: false,
  color: colors.primaryDark,
  height: 'fit-content',
  header: (
    <Text StyledContainer={HeaderText} color="#fff">
      Foundry UI
    </Text>
  ),
  labelFontSize: '.75rem',
  position: 'relative',
};

export default {
  title: 'MainNavigation',
  component: MainNavigation,
  argTypes: {
    position: {
      options: ['relative', 'sticky', 'fixed', 'static'],
      control: {
        type: 'select',
      },
    },
  },
  parameters: {},
} as Meta;
