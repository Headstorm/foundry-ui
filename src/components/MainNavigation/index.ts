import MainNavigation from './MainNavigation';

export default MainNavigation;
