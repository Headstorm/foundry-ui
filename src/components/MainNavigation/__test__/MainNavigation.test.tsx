import React from 'react';
import { render, fireEvent, waitFor, configure } from '@testing-library/react';
import MainNavigation from '../MainNavigation';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);
configure({ testIdAttribute: 'data-test-id' });

const testId = 'hs-ui-main-navigation-unit-test';

describe('MainNavigation', () => {
  it('renders', async () => {
    const { container, getByTestId } = render(
      <MainNavigation containerProps={{ 'data-test-id': testId }} />,
    );

    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('nav button clicks', async () => {
    const { container, getByTestId } = render(
      <MainNavigation
        navButtons={[
          {
            label: 'Test link',
            onClick: () => {},
          },
        ]}
        navButtonProps={[{ 'data-test-id': testId + '-click' }]}
      />,
    );

    await waitFor(() => getByTestId(testId + '-click'));
    fireEvent.mouseDown(getByTestId(testId + '-click'));
    expect(container).toMatchSnapshot();
  });

  describe('Accessibility Tests', () => {
    it('Should pass accessibility test with default props', async () => {
      const component = <MainNavigation />;
      const { container } = render(component);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });
  describe('Ref tests', () => {
    it('containerRef.current should exist', async () => {
      const containerRef = React.createRef<HTMLDivElement>();
      const { getByTestId } = render(
        <MainNavigation
          containerRef={containerRef}
          containerProps={{ 'data-test-id': testId + '-container-ref' }}
        />,
      );
      await waitFor(() => getByTestId(testId + '-container-ref'));
      expect(containerRef.current instanceof HTMLDivElement).toBeTruthy();
    });
    it('headerRef.current should exist', async () => {
      const headerRef = React.createRef<HTMLDivElement>();
      const { getByTestId } = render(
        <MainNavigation
          headerRef={headerRef}
          header={<div></div>}
          headerProps={{ 'data-test-id': testId + '-header-ref' }}
        />,
      );
      await waitFor(() => getByTestId(testId + '-header-ref'));
      expect(headerRef.current instanceof HTMLDivElement).toBeTruthy();
    });
    it('bodyRef.current should exist', async () => {
      const bodyRef = React.createRef<HTMLDivElement>();
      const { getByTestId } = render(
        <MainNavigation
          bodyRef={bodyRef}
          body={<div></div>}
          bodyProps={{ 'data-test-id': testId + '-body-ref' }}
        />,
      );
      await waitFor(() => getByTestId(testId + '-body-ref'));
      expect(bodyRef.current instanceof HTMLDivElement).toBeTruthy();
    });
    it('footerRef.current should exist', async () => {
      const footerRef = React.createRef<HTMLDivElement>();
      const { getByTestId } = render(
        <MainNavigation
          footerRef={footerRef}
          footer={<div></div>}
          footerProps={{ 'data-test-id': testId + '-footer-ref' }}
        />,
      );
      await waitFor(() => getByTestId(testId + '-footer-ref'));
      expect(footerRef.current instanceof HTMLDivElement).toBeTruthy();
    });
  });
});
