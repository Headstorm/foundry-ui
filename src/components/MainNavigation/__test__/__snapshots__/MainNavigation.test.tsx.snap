// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavigation nav button clicks 1`] = `
.c9 {
  opacity: 0;
  background: linear-gradient( 90deg, rgba(255,255,255,0.75), rgba(255,255,255,0.25), rgba(255,255,255,0.75) ) repeat;
  color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-size: 100vw 100vh;
  background-attachment: fixed;
  border-radius: 0.25rem;
  -webkit-animation: gZwmnl 2s linear infinite;
  animation: gZwmnl 2s linear infinite;
}

.c4 {
  display: block;
  position: relative;
}

.c4 > * {
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
}

.c10 {
  position: relative;
}

.c12 {
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  contain: layout;
  outline: 0 none;
  border: 1px solid transparent;
  cursor: pointer;
  background-color: #0000;
  color: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5:hover {
  background-color: rgba(0,0,0,0);
}

.c5:focus {
  outline: none;
  box-shadow: 0 0 5px 0.150rem #338BAA;
}

.c11 {
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}

.c13 {
  border-radius: 0.25em;
}

.c7 {
  font-size: 1rem;
  color: #fff;
}

.c8 {
  display: inline;
}

.c0 {
  position: relative;
  background-color: #BF3A00;
  color: #fff;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  width: 100%;
  -webkit-transform: scaleY(1);
  -ms-transform: scaleY(1);
  transform: scaleY(1);
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  -webkit-transition: -webkit-transform 0.1s cubic-bezier(0,.7,.9,1);
  -webkit-transition: transform 0.1s cubic-bezier(0,.7,.9,1);
  transition: transform 0.1s cubic-bezier(0,.7,.9,1);
}

.c1 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  padding-left: 1rem;
}

.c2 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c3 {
  width: 100%;
}

.c6 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: 0.25rem 0rem 0.25rem 1rem;
  height: 100%;
  border-radius: 0 0 0 0;
}

.c6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6:focus {
  box-shadow: 0 0 0;
}

<div>
  <div
    class=" c0"
    color="#BF3A00"
    height="fit-content"
  >
    <div
      class=" c1"
    >
      <div
        class=" c2 c3"
      >
        <button
          class=" c4 c5 c6"
          color="#0000"
          data-test-id="hs-ui-main-navigation-unit-test-click"
          elevation="0"
          role="button"
          type="button"
        >
          <span
            class=" c7"
            color="#fff"
            data-test-id="hsui-Text"
          >
            <div
              class=" c4 c8"
            >
              <span
                class=" "
              >
                Test link
              </span>
              <div
                class=" c9"
                color="#fff"
              />
            </div>
          </span>
          <div
            class="c10 c11"
          >
            <svg
              class="c12 c13"
              height="0px"
              viewBox="0 0 0 0"
              width="0px"
            />
          </div>
          <div
            class=" c9"
            color="#fff"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`MainNavigation renders 1`] = `
.c0 {
  position: relative;
  background-color: #BF3A00;
  color: #fff;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  width: 100%;
  -webkit-transform: scaleY(1);
  -ms-transform: scaleY(1);
  transform: scaleY(1);
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  -webkit-transition: -webkit-transform 0.1s cubic-bezier(0,.7,.9,1);
  -webkit-transition: transform 0.1s cubic-bezier(0,.7,.9,1);
  transition: transform 0.1s cubic-bezier(0,.7,.9,1);
}

.c1 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  padding-left: 1rem;
}

.c2 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c3 {
  width: 100%;
}

<div>
  <div
    class=" c0"
    color="#BF3A00"
    data-test-id="hs-ui-main-navigation-unit-test"
    height="fit-content"
  >
    <div
      class=" c1"
    >
      <div
        class=" c2 c3"
      />
    </div>
  </div>
</div>
`;
